### PR TITLE
feat(db): implement `ConfigInterface` for `MockDb`

### PR DESCRIPTION
**Pull request by PanGan21**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context

The main motivation is to have MockDb stubs, help to void mocking, and invocation of external database api's.
For more information check https://github.com/juspay/hyperswitch/issues/172

Fixes https://github.com/juspay/hyperswitch/issues/997.

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/db/configs.rs [hunk 1]
```diff
@@ -113,47 +114,110 @@ impl ConfigInterface for Store {
...
-        Err(errors::StorageError::MockDbError)?
+        let configs = self.configs.lock().await;
+        let config = configs.iter().find(|c| c.key == key).cloned();
+
+        match config {
+            Some(c) => Ok(c),
+            None => {
+                Err(errors::StorageError::ValueNotFound("cannot find config".to_string()).into())
+            }
+        }
```
Comment (SanchithHegde) [inline crates/router/src/db/configs.rs:143]: You could use `ok_or_else()` here and elsewhere.

**Review by SanchithHegde — COMMENTED** [review]
Other than that, looks good to me!

File: crates/router/src/db/configs.rs (hunk 1 above)
Comment (PanGan21) [reply crates/router/src/db/configs.rs]: It should be done now! Also fixed the warning of clippy

**Review by SanchithHegde — APPROVED** [review]

**Review by dracarys18 — APPROVED** [review]

**Comment by SanchithHegde** [comment]
Thanks for the PR, @PanGan21!

**Pull request merged**